hack/bazel-generate.sh

```diff
@@ -12,7 +12,7 @@ go_tool_library(
     importmap = "kubevirt.io/kubevirt/vendor/github.com/gordonklaus/ineffassign/pkg/ineffassign",
     importpath = "github.com/gordonklaus/ineffassign/pkg/ineffassign",
     visibility = ["//visibility:public"],
-    deps = ["@org_golang_x_tools//go/analysis:go_tool_library"],
+    deps = ["@org_golang_x_tools//go/analysis:go_default_library"],
```

Comment:
As said, please drop this `cat` command completely and run `make generate` to generate a norma BUILD.bazel file. There is no need for special handling anymore.

Reply:
Hi @rmohr, 
I've performed locally the changes you've asked, but the BUILD.bazel file is not generated under vendor/github.com/gordonklaus/ineffassign/pkg/ineffassign, and `make generate` fails with the following error:

```
ERROR: /root/go/src/kubevirt.io/kubevirt/BUILD.bazel:48:5: no such package 'vendor/github.com/gordonklaus/ineffassign/pkg/ineffassign': BUILD file not found in any of the following directories. Add a BUILD file to a directory to mark it as a package.
 - /root/go/src/kubevirt.io/kubevirt/vendor/github.com/gordonklaus/ineffassign/pkg/ineffassign and referenced by '//:nogo_vet_actual'
```

BTW, it seems like this file has to change as well: `tools/analyzers/BUILD.bazel`

Reply:
@dhiller can you please take a look?

Reply:
@rmohr Hi. For some reason `gazelle` doesn't generate the BUILD.bazel file under the `vendor/github.com/gordonklaus/ineffassign/pkg/ineffassign/` directory. However if we just modify the already commited `BUILD.bazel` with the recent `go_default_library` changes and remove its generation from `generate-bazel.sh` it manage to compile. if we remove the `"gazelle:ignore` annotation from the file, we hit this error: `bazel-out/k8-opt-exec-2B5CBBC6-ST-a03633591ad1/bin/nogo_vet_actual_/nogo_main.go:30:11: cannot use ineffassign.Analyzer (type *"kubevirt.io/kubevirt/vendor/golang.org/x/tools/go/analysis".Analyzer) as type *"golang.org/x/tools/go/analysis".Analyzer in slice literal
compilepkg: error running subcommand external/go_sdk/pkg/tool/linux_amd64/compile: exit status 2
Target @com_github_bazelbuild_buildtools//buildozer:buildozer failed to build
`
I suggest to solve this dependency issue in a follow-up PR, and meanwhile continue using a static BUILD.bazel file as we already do in `tools/analyzers/`

Reply:
@enp0s3 Done.
Thank you for your help!

Reply:
@orelmisan @rmohr by further looking into the issue, I see that in  `bazel-generate.sh` the stage of running `gazelle` will always be after a possible re-generation of the sandbox files. Since the bazeldnf sandboxing depends on nogo, and nogo depends on a vendored analyzer, we cannot run `deps-update` or `deps-prune` right before the sourcing of `bootstrap.sh` because then the BUILD.bazel file will not be in the vendored analyzer. I don't see how the bump of Go 1.17 will resolve that issue, unless I am missing something. @rmohr Please keep me honest here

Reply:
right. I even wrote it back then as a comment :D